freqtrade/optimize/hyperopt_loss_sharpe_daily.py

```diff
@@ -0,0 +1,58 @@
+"""
+SharpeHyperOptLoss
+
+This module defines the alternative HyperOptLoss class which can be used for
+Hyperoptimization.
+"""
+from datetime import datetime
+
+from pandas import DataFrame
+import numpy as np
+
+from freqtrade.optimize.hyperopt import IHyperOptLoss
+
+
+class SharpeHyperOptLossDaily(IHyperOptLoss):
+    """
+    Defines the loss function for hyperopt.
+
+    This implementation uses the Sharpe Ratio calculation.
+    """
+
+    @staticmethod
+    def hyperopt_loss_function(
+        results: DataFrame,
+        trade_count: int,
+        min_date: datetime,
+        max_date: datetime,
+        *args,
+        **kwargs
+    ) -> float:
+        """
+        Objective function, returns smaller number for more optimal results.
+
+        Uses Sharpe Ratio calculation.
+        """
+        total_profit = results.profit_percent
+
+        # adding slippage of 0.1% per trade
+        total_profit = total_profit - 0.0005
+
+        sum_daily = (
+            results.resample("D", on="close_time").agg(
+                {"profit_percent": sum, "profit_abs": sum}
```

Comment:
since we're talking hyperopt - i think `"profit_abs"` should be removed from this dictionary.

It's not used in the remainder of this function - and in hyperopt, every ms you save gets multiplied by epochs ...

Reply:
Good point, let me fix that.